src/mac/mac_ue.c

```diff
@@ -250,8 +250,11 @@ void mac_ue_run_scheduler(MacUE mac) {
             lchan_add_message(chan, msg);
             mac_msg_destroy(msg);
           }
-          MacMessage msg = mac_frag_get_fragment(mac->fragmenter,
-                                                 lchan_unused_bytes(chan), 1);
+          MacMessage msg = NULL;
+          if (mac_frag_has_fragment(mac->fragmenter)) {
```

Comment:
As there is no else criteria here, it might also make sense to include the call of the "mac_frag_has_fragment()" check into mac_frag_get_fragment()?
Otherwise it will probably forgotten to be called before other (future) usages of ...get_fragment() as well.

Reply:
This is a good idea, I've also included this in https://github.com/HAMNET-Access-Protocol/HNAP4PlutoSDR/pull/37/commits/5ffd0417ecce7a9df6af43c2844005a180fcd570